### PR TITLE
Optimizations

### LINKED ISSUE #1759: js-controller 4.0 cannot get logs
grafik
grafik
grafik

### LINKED ISSUE #1766: Error: ENOENT: no such file or directory, open '/opt/iobroker/node_modules/iobroker.web/www/lib/css/them...
Sentry Issue: [IOBROKER-JS-CONTROLLER-296](https://sentry.iobroker.net/organizations/iobroker/issues/24504/?referrer=github_integration)

```
Error: ENOENT: no such file or directory, open '/opt/iobroker/node_modules/iobroker.web/www/lib/css/themes/jquery-ui/hot-sneaks/images/ui-bg_diagonals-small_50_93c3cd_40x40.png'
```

### PR DESCRIPTION
**Pull request by Apollon77**
* Make sure to not change the real logger object but our copy for seq (fixes #1759)
* catch file read errors on upload and log them (fixes #1766)

### CODE REVIEW

File: packages/controller/lib/setup/setupRepo.js [hunk 1]
```diff
@@ -73,18 +78,27 @@ function Repo(options) {
 
         if (urlOrPath.startsWith('http://') || urlOrPath.startsWith('https://')) {
             if (!hash) {
-                hash = await axios({ url: urlOrPath.replace(/\.json$/, '-hash.json'), timeout: 10000 });
+                try {
+                    hash = await axios({ url: urlOrPath.replace(/\.json$/, '-hash.json'), timeout: 10000 });
```
Comment (foxriver76) [inline packages/controller/lib/setup/setupRepo.js:82]: Hash url?

Comment (Apollon77) [reply packages/controller/lib/setup/setupRepo.js:82]: Ääähhm
Yea ;-)) will update

File: packages/controller/lib/setup/setupBackup.js [hunk 2]
```diff
@@ -629,6 +629,9 @@ class BackupRestore {
     }
 
     _copyBackupedFiles(backupDir) {
+        if (!fs.existsSync(backupDir)) {
```
Comment (foxriver76) [inline packages/controller/lib/setup/setupBackup.js:632]: The problem of https://github.com/ioBroker/ioBroker.js-controller/issues/1768 is not, that no path is passed to the method. The problem is, that the path is non existing!

File: packages/controller/lib/setup/setupRepo.js [hunk 3]
```diff
@@ -63,7 +64,11 @@ function Repo(options) {
             oldRepos.native.repositories[repoName].json &&
             (urlOrPath.startsWith('http://') || urlOrPath.startsWith('https://'))
         ) {
-            hash = await axios({ url: urlOrPath.replace(/\.json$/, '-hash.json'), timeout: 10000 });
+            try {
+                hash = await axios({ url: hashUrl, timeout: 10000 });
+            } catch (e) {
+                console.error(`Cannot download repository hash file ${hashUrl}: ${e.message}`);
```
Comment (foxriver76) [inline packages/controller/lib/setup/setupRepo.js:70]:
```suggestion
                console.error(`Cannot download repository hash file from "${hashUrl}": ${e.message}`);
```

File: packages/controller/lib/setup/setupRepo.js [hunk 4]
```diff
@@ -73,18 +78,27 @@ function Repo(options) {
 
         if (urlOrPath.startsWith('http://') || urlOrPath.startsWith('https://')) {
             if (!hash) {
-                hash = await axios({ url: urlOrPath.replace(/\.json$/, '-hash.json'), timeout: 10000 });
+                try {
+                    hash = await axios({ url: urlOrPath.replace(/\.json$/, '-hash.json'), timeout: 10000 });
+                } catch (e) {
+                    console.error(`Cannot download repository hash file ${hashUrl}: ${e.message}`);
```
Comment (foxriver76) [inline packages/controller/lib/setup/setupRepo.js:84]: same as above

File: packages/controller/lib/setup/setupRepo.js [hunk 5]
```diff
@@ -73,18 +78,27 @@ function Repo(options) {
...
+                    timeout: 10000,
+                    headers: { 'User-Agent': agent }
+                });
+                if (data.data) {
+                    data = data.data;
+                } else {
+                    data = null;
+                }
+            } catch (e) {
+                console.error(`Cannot download repository file ${urlOrPath}: ${e.message}`);
```
Comment (foxriver76) [inline packages/controller/lib/setup/setupRepo.js:101]: same

**Review by foxriver76 — CHANGES_REQUESTED** [review]

File: packages/controller/lib/setup/setupBackup.js [hunk 6]
```diff
@@ -629,6 +629,9 @@ class BackupRestore {
...
+            return;
```
Comment (foxriver76) [inline packages/controller/lib/setup/setupBackup.js:633]: is it really okay or should we better log or abort?

Comment (Apollon77) [reply packages/controller/lib/setup/setupBackup.js:633]: I also had the logging idea like "no files to restore". WIll add that

File: packages/controller/lib/setup/setupRepo.js [hunk 7]
```diff
@@ -81,7 +81,7 @@ function Repo(options) {
                 try {
                     hash = await axios({ url: hashUrl, timeout: 10000 });
                 } catch (e) {
-                    console.error(`Cannot download repository hash file ${hashUrl}: ${e.message}`);
```
Comment (foxriver76) [inline packages/controller/lib/setup/setupRepo.js]: also put the hashurl in `"` like first one? :D

**Review by foxriver76 — APPROVED** [review]

**Pull request merged**